src/pages/_app.js

```diff
@@ -145,33 +163,14 @@ function MyApp({ Component, pageProps, intercom, admin, irods, tools }) {
     );
 }
 
-MyApp.getInitialProps = async (ctx) => {
-    const { publicRuntimeConfig = {} } = getConfig() || {};
-    const clientConfig = {
-        intercom: {
-            appId: publicRuntimeConfig.INTERCOM_APP_ID,
-            enabled: publicRuntimeConfig.INTERCOM_ENABLED,
-            companyId: publicRuntimeConfig.INTERCOM_COMPANY_ID,
-            companyName: publicRuntimeConfig.INTERCOM_COMPANY_NAME,
-        },
-        admin: {
-            groups: publicRuntimeConfig.ADMIN_GROUPS,
-            group_attribute_name: publicRuntimeConfig.ADMIN_GROUP_ATTRIBUTE,
-        },
-        irods: {
-            home_path: publicRuntimeConfig.IRODS_HOME_PATH,
-        },
-        tools: {
-            private: {
-                max_cpu_limit: publicRuntimeConfig.TOOLS_PRIVATE_MAX_CPU_LIMIT,
-                max_memory_limit:
-                    publicRuntimeConfig.TOOLS_PRIVATE_MAX_MEMORY_LIMIT,
-                max_disk_limit:
-                    publicRuntimeConfig.TOOLS_PRIVATE_MAX_DISK_LIMIT,
-            },
-        },
-    };
-    return clientConfig;
+MyApp.getInitialProps = async ({ Component, ctx }) => {
+    let pageProps = {};
+
+    if (Component.getInitialProps) {
+        pageProps = await Component.getInitialProps(ctx);
+    }
+
+    return { pageProps, namespacesRequired: ["common"] };
```

Comment:
This is where is spent the most time figuring out. Turns out I need to move the config step into the function and just return pageProps here.